src/library/parser.cpp

```diff
@@ -38,32 +38,31 @@ long Parser::countParsed() {
 }
 
 bool Parser::isBinary(QString filename) {
+    char firstByte;
     QFile file(filename);
-
-    if (file.open(QIODevice::ReadOnly)) {
-        char c;
-        unsigned char uc;
-
-        if(!file.getChar(&c))
-        {
-          qDebug() << "Parser: Error reading stream on " << filename;
-          return true; //should this raise an exception?
+    if (file.open(QIODevice::ReadOnly) && file.getChar(&firstByte)) {
+        qDebug() << "First byte in file:" << static_cast<int>(firstByte);
+        // If starting byte is not an ASCII character then the file
+        // probably contains binary data.
+        if (firstByte >= 32 && firstByte <= 126) {
+            // Valid ASCII character
+            return false;
         }
-
-        uc = uchar(c);
-
-        if(!(33<=uc && uc<=127))  //Starting byte is no character
-        {
-            file.close();
-            return true;
+        // Check for UTF-8 BOM
+        if (firstByte == static_cast<char>(0xEF)) {
+        char nextChar;
```

Comment:
At least at GitHub we have an indent issue.

Reply:
Ok, I need to reformat the changes